extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/output/AzureStorageConnector.java

```diff
@@ -196,7 +196,7 @@ public Iterator<String> listDir(String dirName) throws IOException
     final String prefixBasePath = objectPath(dirName);
     List<String> paths;
     try {
-      paths = azureStorage.listDir(config.getContainer(), prefixBasePath, config.getMaxRetry());
+      paths = azureStorage.listBlobs(config.getContainer(), prefixBasePath, null, config.getMaxRetry());
```

Comment:
Renamed `listDir` to use more specific Azure terminology inside the `Storage` wrapper.